src/main/java/de/dennisguse/opentracks/settings/PreferencesUtils.java

```diff
@@ -54,51 +60,97 @@
 import de.dennisguse.opentracks.ui.customRecordingLayout.RecordingLayoutIO;
 import de.dennisguse.opentracks.util.IntentDashboardUtils;
 
+
+import android.content.SharedPreferences;
+import android.content.res.Resources;
+
+
 /**
  * Utilities to access preferences stored in {@link SharedPreferences}.
  *
  * @author Jimmy Shih
  */
 public class PreferencesUtils {
 
+
     private final static String TAG = PreferencesUtils.class.getSimpleName();
 
+
     private static final int PREFERENCES_VERSION = 2;
 
+
     private PreferencesUtils() {
     }
 
+
     private static SharedPreferences sharedPreferences;
 
+
     private static Resources resources;
 
+
+
+
+
+
+    private static final String SELECTED_COUNTRY_KEY = "selected_country";
+
+
+    public static void saveSelectedCountry(SharedPreferences sharedPreferences, String selectedCountry) {
+        SharedPreferences.Editor editor = sharedPreferences.edit();
+        editor.putString(SELECTED_COUNTRY_KEY, selectedCountry);
+        editor.apply();
+    }
+
+
+    public static String getSelectedCountry(SharedPreferences sharedPreferences) {
+        return sharedPreferences.getString(SELECTED_COUNTRY_KEY, null);
+    }
+
+
+    public static boolean isKey(Resources resources, int resId, String key) {
+        return key.equals(resources.getString(resId));
+    }
+
+
+    public static String getString(Resources resources, int resId) {
+        return resources.getString(resId);
+    }
```

Comment:
These work but this class already has getString and setString functions. You don't need to recreate the getString function, and the saveSelectedCountry and be simplified. See the example:
![image](https://github.com/thanhpd/OpenTracks-Winter-SOEN-6431_2024/assets/6737911/7f523ff6-9a9b-46d3-a408-57b9d2812650)

Your changes should be placed at the bottom, and please config your editor to avoid adding too many new lines in this file.